### DISCUSSION TITLE
Confirmation on update

### DISCUSSION THREAD

**Discussion by bokoch** [filamentphp/filament#4427] [Help]
Can I somehow make updating form with confirmation popup?
I tried to override default save action, like this:
```php
protected function getSaveFormAction(): Action
{
    return Action::make('save')
        ->label(__('filament::resources/pages/edit-record.form.actions.save.label'))
        ->requiresConfirmation()
        ->submit('save')
        ->keyBindings(['mod+s']);
}
```
But it is not shows the confirmation popup and just save the entry. Maybe I need to use `action` method with closure for form submission.

**Comment by bokoch** [comment]
Found such solution worked for me
```php
    protected function getSaveFormAction(): Action
    {
        return Action::make('save')
            ->modalHeading(__('filament.save.modal.heading'))
            ->modalButton(__('filament.save.modal.submit'))
            ->successNotificationMessage(__('filament.save.notification.success.label'))
            ->requiresConfirmation()
            ->action(fn() => $this->save());
    }
```

**Comment by camya** [comment]
I tried it too , but it does not work for me. (see screenshot & source code below) 

The confirmations modal opens, but also the validation is triggered. The modal should not open, if there are any errors.



<img width="873" alt="Screenshot 2022-10-10 at 00 25 08" src="https://user-images.githubusercontent.com/827658/194782277-bc2957d6-55e3-4d6a-bb1b-7c15276bc39a.png">

! Mentioned source code: Moved new version in a later comment, see below.

**Reply by danharrin** [reply]
You need to remove the `submit()` I think

**Reply by camya** [reply]
Hi, 

If I remove the submit() method, the Confirmation modal opens fine... **unfortunately without triggering the validation** first. (Image 1)

Only if I click "Confirm" in the modal, the validation is triggered and the errors are shown. (Image 2)

Maybe another parameter is missing?

# Image 1
<img width="506" alt="Screenshot 2022-10-10 at 09 30 55" src="https://user-images.githubusercontent.com/827658/194817068-317a9fa9-7496-4dc6-a032-f44de0b62f2c.png">

# Image 2 (Confirm clicked)
<img width="498" alt="Screenshot 2022-10-10 at 09 31 03" src="https://user-images.githubusercontent.com/827658/194817062-5c578900-f812-4972-ace6-895a5f85accc.png">

**Reply by danharrin** [reply]
Try `->mountUsing(fn () => $this->form->validate())`

**Reply by camya** [reply]
It works now... great! That mountUsing closure was the missing part. :) 

Thank you for the hint. 

Below the working code. (Only ->keyBindings(['mod+s']) does nothing on my Mac M1)

```
<?php

namespace App\Filament\Resources\PostResource\Pages;

use App\Filament\Resources\PostResource;
use Filament\Pages\Actions\Action;
use Filament\Resources\Pages\CreateRecord;
use Illuminate\Support\HtmlString;

class CreatePost extends CreateRecord
{
    protected static string $resource = PostResource::class;

    protected function getCreateFormAction(): Action
    {
        return Action::make('create')
            ->requiresConfirmation()
            ->mountUsing(fn() => $this->form->validate())
            ->action(fn() => $this->create())
            ->keyBindings(['mod+s'])
            ->modalHeading('Save this article?')
            ->modalSubheading(function () {

                $publishDate = $this->data['published_at']
                    ? carbon($this->data['published_at'])->format("d.m.Y H:i")
                    : ' -- ';

                return new HtmlString('
                    <div class="w-full text-left">
                        <ul class="text-sm space-y-2">
                            <li><strong>Title:</strong> '.e($this->data["title"]).'</li>
                            <li><strong>Slug:</strong> '.e($this->data["slug"]).'</li>
                            <li><strong>Publish date:</strong> '.e($publishDate).'</li>
                        </ul>
                    </div>
                ');
            });
    }

}
```

**Reply by sofianegargouri** [reply]
Hi mate,
Tried this code, kind of works at first. Unfortunately I have the following issue: when I have an error in my form and submit it, the validation is well triggered, but it isn't retriggered if I reclick on the submit button without changing anything and just opens the modal

**Comment by liquid207** [comment]
@danharrin Thanks a lot, save my day
